client/src/state/editor/sortBlockMutation.js

```diff
@@ -20,6 +21,42 @@ const config = {
         blockId,
         afterBlockId,
       },
+      optimisticResponse: {
+        ID: blockId,
+        __typename: 'Block',
+      },
+      update: store => {
+        const variables = readBlocksConfig.options({ pageId: 6 }).variables; // Fix me
+        const data = store.readQuery({ query: readBlocksQuery, variables });
+
+        // Query returns a deeply nested object. Explicit reconstruction via spreads is too verbose.
+        // This is an alternative, relatively efficient way to deep clone
+        const newData = JSON.parse(JSON.stringify(data));
+        let { edges } = newData.readOnePage.ElementalAreaIfExists.Elements;
+
+        // Find the block we reordered
+        const movedBlockIndex = edges.findIndex(edge => edge.node.ID === blockId);
+        // Keep it
+        const movedBlock = edges[movedBlockIndex];
+        // Remove the moved block
+        edges.splice(movedBlockIndex, 1);
+        // If the target is 0, it's added to the start
+        if (!afterBlockId) {
+          edges.unshift(movedBlock);
+        } else {
+          // Else, find the block we inserted after
+          const targetBlockIndex = edges.findIndex(edge => edge.node.ID === afterBlockId);
+          // Add it back after the target
+          const end = edges.slice(targetBlockIndex + 1);
+          edges = edges.slice(0, targetBlockIndex + 1);
+          edges.push(movedBlock);
+          edges = edges.concat(end);
+        }
+
+        // Add it back to the full result
+        newData.readOnePage.ElementalAreaIfExists.Elements.edges = edges;
+        store.writeQuery({ query: readBlocksQuery, data: newData, variables });
```

Comment:
Cool 😄

Reply:
I finished the implementation of this. Now the function takes `pageId` as a parameter (because it needs to know what `readBlocksForPage` query to update). The params are different from the query but that's fine I assume.